server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -181,6 +180,11 @@ public boolean supportsVersion(Version indexCreatedVersion) {
         @Override
         public Mapper.Builder parse(String name, Map<String, Object> node, MappingParserContext parserContext)
             throws MapperParsingException {
+            if (parserContext instanceof MappingParserContext.ObjectParserContext objectParserContext) {
```

Comment:
Rather than these instanceof checks, how about just adding the depth directly to MappingParserContext?

Reply:
@romseygeek Thanks for the feedback. I attempted this implementation in f38127dafe3fd50af33aad11ba0087bcc128db73, but I like this version less, as we need to introduce `ThreadLocal` variables, as the same `MappingParserContext` can be shared by several threads. WDYT?

Reply:
@romseygeek Thanks for the suggestion, I've addressed it in 28abd2ff49cfa0673a681ffa35a96aef1335d62c